assets/js/blocks/featured-product/block.js

```diff
@@ -339,6 +341,30 @@ const FeaturedProduct = ( {
 											} )
 										}
 									/>
+									<TextareaControl
+										label={ __(
+											'Alt text (alternative text)',
+											'woo-gutenberg-products-block'
+										) }
+										value={ attributes.alt }
+										onChange={ ( newAlt ) => {
+											setAttributes( { alt: newAlt } );
+										} }
```

Comment:
Super nitpick, but why not:

```suggestion
										onChange={ ( alt ) => {
											setAttributes( { alt } );
										} }
```